logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/ValidatePasswordUseCase.kt

```diff
@@ -54,5 +53,24 @@ internal class ValidatePasswordUseCaseImpl : ValidatePasswordUseCase {
                 ".{$PASSWORD_MIN_LENGTH,}" + // min PASSWORD_MIN_LENGTH characters
                 "$"
                 ).toRegex()
+        private val PASSWORD_LOWERCASE_REGEX = "^.*[a-z].*$".toRegex() // at least one lowercase ASCII letter
+        private val PASSWORD_UPPERCASE_REGEX = "^.*[A-Z].*$".toRegex() // at least one uppercase ASCII letter
+        private val PASSWORD_DIGIT_REGEX = "^.*[0-9].*$".toRegex() // at least a digit
+        private val PASSWORD_SPECIAL_CHAR_REGEX = "^.*[^a-zA-Z0-9].*$".toRegex() // at least a "special character"
+        private val PASSWORD_LENGTH_REGEX = "^.{$PASSWORD_MIN_LENGTH,}$".toRegex() // min PASSWORD_MIN_LENGTH characters
     }
 }
+
+sealed class ValidatePasswordResult {
+    data object Valid : ValidatePasswordResult()
+    data class Invalid(
+        val missingLowercaseCharacter: Boolean = true,
+        val missingUppercaseCharacter: Boolean = true,
+        val missingDigit: Boolean = true,
+        val missingSpecialCharacter: Boolean = true,
+        val tooShort: Boolean = true,
```

Comment:
thought: maybe use a `Set<Criteria>` instead of multiple fields.

For example:
```kotlin
enum class Criteria {
    LOWERCASE, UPPERCASE, NUMERIC_DIGIT, SPECIAL_CHARACTER, LENGTH;
}

sealed interface ValidatePasswordResult {
    data object Valid: ValidatePasswordResult()
    data class Invalid(val missingCriteria: Set<Criteria>)
}
```

It would be easier to scale and change in the future:

```kotlin
if(result.contains(LOWERCASE)) {
    // Add Lowercase warning thing
}
if(result.contains(SPECIAL_CHARACTER)) {
    // Add special character warning thing
}
```

----

In any case, not something I'd change now that it's done. Just food for thought really. Aiming to make the API a tiny bit more scalable.